--- xvalg.c
+++ xvalg.c
@@ -54,8 +54,6 @@ int         start24bitAlg  PARM((byte **
 void        end24bitAlg    PARM((byte *, byte *));
 void        printUTime     PARM((const char *));
 
-byte *origPic = (byte *) NULL;
-int  origPicType;
 static byte origrmap[256], origgmap[256], origbmap[256];
 
 
--- xv.c
+++ xv.c
@@ -214,7 +214,7 @@ int main(argc, argv)
   visualstr = monofontname = flistName = NULL;
   winTitle = NULL;
 
-  pic = egampic = epic = cpic = NULL;
+  pic = egampic = epic = cpic = origPic = NULL;
   theImage = NULL;
 
   picComments = (char *) NULL;
--- xv.h
+++ xv.h
@@ -1183,6 +1183,8 @@ WHERE int            picType;
 WHERE char          *picComments;           /* text comments on current pic */
 WHERE byte          *picExifInfo;           /* image info from digicam */
 WHERE int            picExifInfoSize;       /* size of image info */
+WHERE byte          *origPic;               /* ptr to original picture */
+WHERE int            origPicType;           /* type of original picture */
 
 #ifdef TV_L10N
 WHERE int            xlocale;		    /* true if Xlib supports locale */
--- xvimage.c
+++ xvimage.c
@@ -348,9 +348,6 @@ int w,h;
   CreateXImage();
 }
 
-byte *origPic;
-int  origPicType;
-
 /********************************************/
 void GenerateCpic()
 {
--- xvmask.c
+++ xvmask.c
@@ -42,8 +42,6 @@ void end24bitAlg 		PARM ((byte *, byte *
 void saveOrigPic 		PARM ((void));
 void printUTime 		PARM ((char *));
 
-byte *origPic;
-
 #undef TIMING_TEST
 
 #ifdef TIMING_TEST
